### PR TITLE
[BugFix] Multi bytes CSV row/column delimiter for `files()`

### PR DESCRIPTION
**Pull request by rickif**
## Why I'm doing:
Now, `files` cannot handle CSV multi-byte delimiter correctly. 

## What I'm doing:
Fixes https://github.com/StarRocks/StarRocksTest/issues/7268
This PR including following changes:
1. It corrects the CSV format multi bytes delimiter for `files()`
2. It improves the error message for the delimiter longer than 50 bytes. The new error message will be  `The valid bytes length for 'csv.row_delimiter' is [1, 50]` and `The valid bytes length for 'csv.column_separator' is [1, 50]`.

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/catalog/TableFunctionTable.java [hunk 1]
```diff
@@ -263,10 +264,18 @@ private void parseProperties() throws DdlException {
 
         if (properties.containsKey(PROPERTY_CSV_COLUMN_SEPARATOR)) {
             csvColumnSeparator = properties.get(PROPERTY_CSV_COLUMN_SEPARATOR);
+            int len  = csvColumnSeparator.getBytes(StandardCharsets.UTF_8).length;
```
Comment (rickif) [inline fe/fe-core/src/main/java/com/starrocks/catalog/TableFunctionTable.java:267]:
```suggestion
            int len = csvColumnSeparator.getBytes(StandardCharsets.UTF_8).length;
```

File: fe/fe-core/src/main/java/com/starrocks/catalog/TableFunctionTable.java [hunk 2]
```diff
@@ -263,10 +264,18 @@ private void parseProperties() throws DdlException {
 
         if (properties.containsKey(PROPERTY_CSV_COLUMN_SEPARATOR)) {
             csvColumnSeparator = properties.get(PROPERTY_CSV_COLUMN_SEPARATOR);
+            int len = csvColumnSeparator.getBytes(StandardCharsets.UTF_8).length;
+            if (len > 50 || len == 0) {
+                throw new DdlException("the length of csv.column_separator property should be in [1, 50]");
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/catalog/TableFunctionTable.java:269]: the error message is `xxx is limited to a maximum of 50 bytes` in `StreamLoadScanNode` and `FileScanNode` and  better use the same error message.

put the error message in `ErrorCode`?

File: fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java [hunk 3]
```diff
@@ -302,6 +302,7 @@ public enum ErrorCode {
             "Referenced column '%s' in expr '%s' can't be found in column list, derived column is '%s'"),
     ERR_MAPPING_EXPR_INVALID(5602, new byte[] {'4', '2', '0', '0', '0'}, "Expr '%s' analyze error: %s, derived column is '%s'"),
 
+    ERR_VALUE_OUT_OF_VALID_RANGE(5605, new byte[] {'4', '2', '0', '0', '0'}, "The valid range of values for '%s' is [%d, %d]"),
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java:305]:
```suggestion
    ERR_VALUE_OUT_OF_VALID_RANGE(5605, new byte[] {'4', '2', '0', '0', '0'}, "The valid bytes length for '%s' is [%d, %d]"),
```

File: fe/fe-core/src/main/java/com/starrocks/catalog/TableFunctionTable.java [hunk 4]
```diff
@@ -263,10 +264,20 @@ private void parseProperties() throws DdlException {
 
         if (properties.containsKey(PROPERTY_CSV_COLUMN_SEPARATOR)) {
             csvColumnSeparator = properties.get(PROPERTY_CSV_COLUMN_SEPARATOR);
+            int len = csvColumnSeparator.getBytes(StandardCharsets.UTF_8).length;
+            if (len > 50 || len == 0) {
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/catalog/TableFunctionTable.java:268]: better define a variable `50` in a common place

File: fe/fe-core/src/main/java/com/starrocks/common/CsvFormat.java [hunk 5]
```diff
@@ -16,6 +16,8 @@
 package com.starrocks.common;
 
 public class CsvFormat {
+    public static final int MAX_COLUMN_SEPARATOR_LENGTH = 50;
+    public static final int MAX_ROW_DELIMITER_LENGTH = 50;
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/common/CsvFormat.java:20]: add a blank line

File: fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java [hunk 6]
```diff
@@ -302,6 +302,7 @@ public enum ErrorCode {
             "Referenced column '%s' in expr '%s' can't be found in column list, derived column is '%s'"),
     ERR_MAPPING_EXPR_INVALID(5602, new byte[] {'4', '2', '0', '0', '0'}, "Expr '%s' analyze error: %s, derived column is '%s'"),
 
+    ERR_ILLEGAL_BYTES_LENGTH(5605, new byte[] {'4', '2', '0', '0', '0'}, "The valid bytes length for '%s' is [%d, %d]"),
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java:305]: add a blank line

**Review by wyb — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/common/CsvFormat.java (hunk 5 above)
Comment (rickif) [inline fe/fe-core/src/main/java/com/starrocks/common/CsvFormat.java:20]:
```suggestion
    public static final int MAX_ROW_DELIMITER_LENGTH = 50;
    
```

**Review by meegoo — APPROVED** [review]

**Review by yingtingdong — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Review by HangyuanLiu — APPROVED** [review]

**Comment by rickif** [comment]
@Mergifyio backport branch-3.2

**Pull request merged**